src/lib/blockchain_snark/blockchain_state.ml

```diff
@@ -82,10 +96,63 @@ module Make (Consensus_mechanism : Consensus.S) :
               ( transition |> Snark_transition.blockchain_state
               |> Blockchain_state.snarked_ledger_hash )
           in
+          let%bind new_pending_coinbase_hash =
+            let prev_root =
+              previous_state |> Protocol_state.blockchain_state
+              |> Blockchain_state.pending_coinbase_hash
+            in
+            let updated_stack = pending_coinbase_update.updated_stack in
+            let action = pending_coinbase_update.action in
+            let with_del_add del_added =
+              Pending_coinbase.Checked.update_delete_stack
+                ~is_new_stack:Boolean.true_ ~stack:updated_stack prev_root
+            in
+            let with_del_update del_updated =
+              Pending_coinbase.Checked.update_delete_stack
+                ~is_new_stack:Boolean.false_ ~stack:updated_stack prev_root
+            in
+            let with_add added =
+              Pending_coinbase.Checked.update_stack prev_root
+                ~is_new_stack:Boolean.true_ updated_stack
+            in
+            let with_update updated =
+              Pending_coinbase.Checked.update_stack prev_root
+                ~is_new_stack:Boolean.false_ updated_stack
+            in
+            let%bind added =
+              Pending_coinbase_update.Action.Checked.added action
+            in
+            let%bind updated =
+              Pending_coinbase_update.Action.Checked.updated action
+            in
+            let%bind del_add =
+              Pending_coinbase_update.Action.Checked.deleted_added action
+            in
+            let%bind del_updated =
+              Pending_coinbase_update.Action.Checked.deleted_updated action
+            in
+            let chain if_ b ~then_ ~else_ =
+              let%bind then_ = then_ and else_ = else_ in
+              if_ b ~then_ ~else_
+            in
+            chain Pending_coinbase.Hash.if_ added ~then_:(with_add added)
+              ~else_:
+                (chain Pending_coinbase.Hash.if_ updated
+                   ~then_:(with_update updated)
+                   ~else_:
+                     (chain Pending_coinbase.Hash.if_ del_add
+                        ~then_:(with_del_add del_add)
+                        ~else_:(with_del_update del_updated)))
+          in
+          let%bind correct_coinbase_status =
+            let new_root = pending_coinbase_update.new_root in
+            Pending_coinbase.Hash.equal_var new_pending_coinbase_hash new_root
+          in
           let%bind correct_snark =
```

Comment:
I think this is might now be broken because there is other info we use from the SNARK besides the ledge hash (e.g., total supply etc.).

It would probably be better if we first compute `correct_snark` and then use it as a mask on all the fields that we update using data from the input of the ledger proof.